flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/BigQueryTableInfo.java

```diff
@@ -33,4 +34,24 @@ public static TableSchema getSchema(
                                                 "The provided table %s.%s.%s does not exists.",
                                                 project, dataset, table)));
     }
+
+    /**
+     * Function to identify if a BigQuery table exists.
+     *
+     * @param client {@link BigQuery} Object containing the BigQuery Client.
+     * @param dataset Dataset ID containing the Table.
+     * @param tableName Table Name.
+     * @return Boolean {@code TRUE} if the table exists or {@code FALSE} if it does not.
+     */
+    public static Boolean tableExists(BigQuery client, String dataset, String tableName) {
+        try {
+            Table table = client.getTable(TableId.of(dataset, tableName));
+            return (table != null && table.exists());
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "The provided Bigquery table %s.%s not found with exception: %s",
+                            dataset, tableName, e.toString()));
+        }
```

Comment:
1. We should configure retries in the client for retry-able errors.
2. `IllegalArgumentException` is not right here. Throw `BigQueryConnectorException`.
3. Error thrown should be `throw new BigQueryConnectorException(String.format("Could not determine existence of BigQuery table %s.%s.%s", projectName, datasetName, tableName), e)`

Reply:
I just checked, it seems that BigQuery client has default retry settings. Please verify the same. If true, then we dont need to configure it ourselves.

Reply:
Yes, some default retry strategy is already configured by BigQuery default builder as:

```
RetrySettings.newBuilder()
        .setMaxAttempts(6)
        .setInitialRetryDelay(Duration.ofMillis(1000L))
        .setMaxRetryDelay(Duration.ofMillis(32_000L))
        .setRetryDelayMultiplier(2.0)
        .setTotalTimeout(Duration.ofMillis(50_000L))
        .setInitialRpcTimeout(Duration.ofMillis(50_000L))
        .setRpcTimeoutMultiplier(1.0)
        .setMaxRpcTimeout(Duration.ofMillis(50_000L))
```